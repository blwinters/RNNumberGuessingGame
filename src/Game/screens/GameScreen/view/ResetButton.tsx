import React from 'react'
import {Button} from 'react-native'
import {CheckGuessResult} from 'src/Game/types'

interface Props {
  result: CheckGuessResult | null
  onPress: () => void
}

const ResetButton = ({result, onPress}: Props) => {
  if (result === 'correct') {
    return <Button title="New Game" onPress={onPress} />
  } else {
    return <></>
  }
}

export default ResetButton
