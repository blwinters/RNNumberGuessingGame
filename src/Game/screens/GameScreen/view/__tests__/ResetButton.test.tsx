import React from 'react'
import {render, RenderAPI} from 'react-native-testing-library'
import ResetButton from '../ResetButton'

describe('ResetButton.tsx', () => {
  let instance: RenderAPI

  describe('when the result is `correct`', () => {
    beforeEach(() => {
      instance = render(<ResetButton result="correct" onPress={() => {}} />)
    })
    it('should match snapshot', () => {
      expect(instance.toJSON()).toMatchSnapshot()
    })
  })

  describe('when the result is `tooLow`', () => {
    beforeEach(() => {
      instance = render(<ResetButton result="tooLow" onPress={() => {}} />)
    })
    it('should match snapshot', () => {
      expect(instance.toJSON()).toMatchSnapshot()
    })
  })

  describe('when the result is `tooHigh`', () => {
    beforeEach(() => {
      instance = render(<ResetButton result="tooHigh" onPress={() => {}} />)
    })
    it('should match snapshot', () => {
      expect(instance.toJSON()).toMatchSnapshot()
    })
  })
})
