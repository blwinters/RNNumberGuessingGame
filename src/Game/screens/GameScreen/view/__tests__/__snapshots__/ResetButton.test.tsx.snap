// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetButton.tsx when the result is \`correct\` should match snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      New Game
    </Text>
  </View>
</View>
`;

exports[`ResetButton.tsx when the result is \`tooHigh\` should match snapshot 1`] = `null`;

exports[`ResetButton.tsx when the result is \`tooLow\` should match snapshot 1`] = `null`;
